sdks/go/pkg/beam/core/runtime/exec/datasource.go

```diff
@@ -320,6 +321,73 @@ func (n *DataSource) Progress() ProgressReportSnapshot {
 	return ProgressReportSnapshot{ID: n.SID.PtransformID, Name: n.Name, Count: c, pcol: pcol}
 }
 
+func (n *DataSource) getProcessContinuation() sdf.ProcessContinuation {
+	if u, ok := n.Out.(*ProcessSizedElementsAndRestrictions); ok {
```

Comment:
Consider adding a comment here that current support requires the SDF to be immediately after the datasource node.  This is true for most runners (it's easiest to reason about), but it's not a requirement of the model. See also how we could theoretically support multiple Datasources in a single bundled, but in practice, that never happens....